lapis/utility/monitor.py

```diff
@@ -22,94 +43,197 @@ def __init__(self, simulator: "Simulator"):
+            "allocation": drone.allocation,
+            "utilisation": drone.utilisation,
+            "demand": drone.demand,
+            "supply": drone.supply,
+            "job_count": drone.jobs
+        })
+    return results
+
+
+collect_drone_cobald_statistics.name = "cobald_status"
+collect_drone_cobald_statistics.logging_formatter = {
+    LoggingSocketHandler.__name__: JsonFormatter(),
+    logging.StreamHandler.__name__: JsonFormatter(),
+    LoggingUDPSocketHandler.__name__: LineProtocolFormatter(
+        tags={"tardis", "pool_configuration", "pool_type", "pool"},
+        resolution=1
+    )
+}
+
+
+def collect_pool_cobald_statistics(simulator: "Simulator") -> list:
+    """
+    Function takes care on collecting statistics of pools, i.e. allocation,
+    utilisation, demand, and supply.
+
+    :param simulator: the simulator
+    :return: list of records to log
+    """
+    results = []
     for pool in simulator.pools:
-        pool_demand[repr(pool)] = pool.demand
-        pool_supply[repr(pool)] = pool.supply
-        pool_utilisation[repr(pool)] = pool.utilisation
-        pool_allocation[repr(pool)] = pool.allocation
-    return {
-        "pool": {
-            "demand": pool_demand,
-            "supply": pool_supply,
-            "allocation": pool_allocation,
-            "utilisation": pool_utilisation
-        }
-    }
+        results.append({
+            "pool_configuration": "None",
+            "pool_type": "pool",
+            "pool": repr(pool),
+            "allocation": pool.allocation,
+            "utilisation": pool.utilisation,
+            "demand": pool.demand,
+            "supply": pool.supply,
+        })
+    return results
+
+
+collect_pool_cobald_statistics.name = "cobald_status"
+collect_pool_cobald_statistics.logging_formatter = {
+    LoggingSocketHandler.__name__: JsonFormatter(),
+    logging.StreamHandler.__name__: JsonFormatter(),
+    LoggingUDPSocketHandler.__name__: LineProtocolFormatter(
+        tags={"tardis", "pool_configuration", "pool_type", "pool"},
+        resolution=1
+    )
+}
+
+
+def collect_pool_status(simulator: "Simulator") -> list:
+    """
+    Function takes care on logging information about when pools and drones
+    did change state within the system, e.g. were integrated or removed.
+
+    :param simulator: the simulator
+    :return: list of records for logging
+    """
+    return []
+
+
+collect_pool_status.name = "pool_status"
+collect_pool_status.logging_formatter = {
+    LoggingSocketHandler.__name__: JsonFormatter(),
+    logging.StreamHandler.__name__: JsonFormatter(),
+    LoggingUDPSocketHandler.__name__: LineProtocolFormatter(
+        tags={"tardis", "parent_pool", "pool_configuration", "pool_type", "pool"},
+        resolution=1
+    )
+}
+
+
+def collect_configuration_information(simulator: "Simulator") -> list:
+    """
+    Function takes care on logging information about the configuration of
+    pools and drones, e.g. provided resources.
```

Comment:
How about "Log state changes of pools and drones"?

Reply:
Not for this one, but the one above.